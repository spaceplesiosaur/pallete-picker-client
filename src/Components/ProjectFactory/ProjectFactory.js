import React from 'react';
import Project from '../Project/Project';
import './ProjectFactory.scss';

import PropTypes from 'prop-types';

const ProjectFactory = ({ projects, palettes, removeProject, fetchPalettes }) => {
  const generateProjects = () => {
    return projects.map(project => {
      const matchingPalettes = palettes.filter(palette => {
          return palette.project_id === project.id
        })

      return (
        <Project
          key={project.id}
          name={project.name}
          id={project.id}
          palettes={matchingPalettes}
          removeProject={removeProject}
          fetchPalettes={fetchPalettes}
        />
      )
    })
  }
  return (
    <section className="project-factory-renderArea">
      {generateProjects()}
    </section>
  )
};

ProjectFactory.propTypes = {
  projects: PropTypes.array,
  palettes: PropTypes.array,
  removeProject: PropTypes.func.
  fetchPalettes: PropTypes.func
};

export default ProjectFactory
