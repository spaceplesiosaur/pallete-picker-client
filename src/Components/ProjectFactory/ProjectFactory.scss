.project-factory-renderArea {
    display: flex;
    flex-wrap: wrap;
    height: 35em;
    overflow: scroll;
}