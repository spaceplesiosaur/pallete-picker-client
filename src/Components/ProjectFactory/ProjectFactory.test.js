import React from 'react';
import { shallow } from 'enzyme';
import ProjectFactory from './ProjectFactory'

describe('ProjectFactory', () => {
  let wrapper;
  let mockProjects;
  let mockPalettes;

  beforeEach(() => {

    mockProjects = [
  {
    id: 1,
    name: "Bathroom walls",
    current: true,
  },
  {
    id: 2,
    name: 'Kid bedroom',
    current: false,
  },
  {
    id: 3,
    name: 'Living room',
    current: false,
  }
]

    mockPalettes = [
  {
    id: 4,
    name: "Ocean",
    color1: "#DFE9FD",
    color2: "#053BA7",
    color3: "#B3C0F7",
    color4: "#4B93FB",
    color5: "#00A8CF",
    project_id: 1
  },
  {
    id: 5,
    name: "Seascape",
    color1: "#180353",
    color2: "#00A8CF",
    color3: "#F7EDB7",
    color4: "#EDE0D7",
    color5: "#DFE9FD",
    project_id: 1
  },
  {
    id: 6,
    name: "Sunset",
    color1: "#FA1E88",
    color2: "#ff6100",
    color3: "#F8DF3A",
    color4: "#B3C0F7",
    color5: "#000000",
    project_id: 2
  }
]
    wrapper = shallow(<ProjectFactory
      projects={mockProjects}
      palettes={mockPalettes}
      removeProject={jest.fn()}
      />)

  })

  it('should render with the correct properites', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
