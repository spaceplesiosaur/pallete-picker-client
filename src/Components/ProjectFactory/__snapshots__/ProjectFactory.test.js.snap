// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectFactory should render with the correct properites 1`] = `
<section
  className="project-factory-renderArea"
>
  <Project
    fetchPalettes={[MockFunction]}
    id={1}
    key="1"
    name="Bathroom walls"
    palettes={
      Array [
        Object {
          "color1": "#DFE9FD",
          "color2": "#053BA7",
          "color3": "#B3C0F7",
          "color4": "#4B93FB",
          "color5": "#00A8CF",
          "id": 4,
          "name": "Ocean",
          "project_id": 1,
        },
        Object {
          "color1": "#180353",
          "color2": "#00A8CF",
          "color3": "#F7EDB7",
          "color4": "#EDE0D7",
          "color5": "#DFE9FD",
          "id": 5,
          "name": "Seascape",
          "project_id": 1,
        },
      ]
    }
    removeProject={[MockFunction]}
  />
  <Project
    fetchPalettes={[MockFunction]}
    id={2}
    key="2"
    name="Kid bedroom"
    palettes={
      Array [
        Object {
          "color1": "#FA1E88",
          "color2": "#ff6100",
          "color3": "#F8DF3A",
          "color4": "#B3C0F7",
          "color5": "#000000",
          "id": 6,
          "name": "Sunset",
          "project_id": 2,
        },
      ]
    }
    removeProject={[MockFunction]}
  />
  <Project
    fetchPalettes={[MockFunction]}
    id={3}
    key="3"
    name="Living room"
    palettes={Array []}
    removeProject={[MockFunction]}
  />
</section>
`;
